openquake/baselib/node.py

```diff
@@ -314,7 +314,19 @@ def serialize(self, node):
             return
         self.start_tag(tag, node.attrib)
         if node.text is not None:
-            txt = escape(scientificformat(node.text).strip())
+            if node.tag == 'posList':
+                # NOTE: by convention, posList must be a flat list of
+                #       space-separated coordinates.
+                #       Here we expect as input a list of lists of tuples
```

Comment:
Not clear why you are considering text[0] only. Has the list always a single element?

Reply:
I am now flattening any tuples or lists, regardless from the nesting.